### PR TITLE
Third home work

### PR DESCRIPTION
**Pull request by MuravskyVadim**

### CODE REVIEW

File: src/ComplexNumber.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        return re;
+    }
+
+    public double getIm() {
+        return im;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
```
Comment (Sofasmile) [inline src/ComplexNumber.java:20]: cs

File: src/ComplexNumber.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        if (obj instanceof ComplexNumber) {
```
Comment (Sofasmile) [inline src/ComplexNumber.java:22]: where is the checking of the null?

File: src/ThirdHomeWork.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                "Городничий: Городничий: Ревизор! Ревизор! Ревизор!",
+                "Лука: ... с секретным предписаньем!"};
+
+        System.out.print(printTextPerRole(roles, textLines));
+    }
+
+    public static int[] mergeArrays(int[] firstArray, int[] secondArray) {
+        int arrayLength = firstArray.length + secondArray.length;
+        int[] resultArray = new int[arrayLength];
+        int i = 0, j = 0, k = 0;
```
Comment (Sofasmile) [inline src/ThirdHomeWork.java]: cs

File: src/ThirdHomeWork.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        while (i < firstArray.length || j < secondArray.length) {
```
Comment (Sofasmile) [inline src/ThirdHomeWork.java]: optional: you can realize this method more simply

File: src/ThirdHomeWork.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        for (int i = 0; i < roles.length; i++) {
+            String role = roles[i] + ":";
+            result.append(role);
+
+            for (int j = 0; j < textLines.length; j++) {
+                if (textLines[j].startsWith(role)) {
+                    result.append("\n").append(j + 1).append(") ").append(textLines[j].substring(role.length() + 1));
+                }
+
+                if (j == textLines.length - 1 && i != roles.length - 1) {
```
Comment (Sofasmile) [inline src/ThirdHomeWork.java:74]: you can remove this if statement and add before 
if (i < roles.length - 1) {
                result.append("\n");
            }
result.append("\n");

**Review by Sofasmile — COMMENTED** [review]
resolve comments

File: src/ComplexNumber.java (hunk 2 above)
Comment (MuravskyVadim) [reply src/ComplexNumber.java:22]: "instanceof" checked of the null, but if you want I may add checking obviously "if (obj != null && obj instanceof ComplexNumber)"

File: src/ThirdHomeWork.java (hunk 4 above)
Comment (MuravskyVadim) [reply src/ThirdHomeWork.java]: Ok. I did it.

File: src/ThirdHomeWork.java (hunk 5 above)
Comment (MuravskyVadim) [reply src/ThirdHomeWork.java:74]: doesn't work with this replacement or I don't understand where it is necessary to make a replacement

File: src/ComplexNumber.java (hunk 2 above)
Comment (Sofasmile) [reply src/ComplexNumber.java:22]: no need to add checking. It is ok

File: src/ThirdHomeWork.java (hunk 5 above)
Comment (Sofasmile) [reply src/ThirdHomeWork.java:74]: private static String printTextPerRole(String[] roles, String[] textLines) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < roles.length; i++) {
            String role = roles[i] + ":";
            result.append(role);
            for (int j = 0; j < textLines.length; j++) {
                if (textLines[j].startsWith(role)) {
                    result.append("\n").append(j + 1).append(") ").append(textLines[j].substring(role.length() + 1));
                }
            }
            result.append("\n");
            if (i < roles.length - 1) {
                result.append("\n");
            }
        }
        return result.toString();
    }

**Review by Sofasmile — APPROVED** [review]
good, make small fixing

**Pull request merged**